### ISSUE TITLE
geometry.asShape with empty geometries

### ISSUE THREAD

**Issue by jahight** [Toblerity/Shapely#542]
I have run into an issue when using `geometry.asShape()` with empty geometries as input.  Calling `.is_empty` on the returned object raises AssertionError.  Using `geometry.shape()` functions as expected.

```python
Python 3.6.2 (default, Aug  7 2017, 17:26:39) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import shapely
>>> shapely.__version__
'1.6.2.post1'
>>> from shapely import geometry
>>> empty_mp = geometry.MultiPolygon()
>>> empty_mp.is_empty
True
>>> empty_json = geometry.mapping(empty_mp)
>>> print(empty_json)
{'type': 'MultiPolygon', 'coordinates': []}
>>> empty_shape = geometry.shape(empty_json)
>>> empty_shape.is_empty
True
>>> empty_asShape = geometry.asShape(empty_json)
>>> empty_asShape.is_empty
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/python-3.6/lib/python3.6/site-packages/shapely/geometry/base.py", line 643, in is_empty
    return (self._geom is None) or bool(self.impl['is_empty'](self))
  File "/opt/python-3.6/lib/python3.6/site-packages/shapely/geometry/proxy.py", line 33, in _geom
    self.__geom__, n = self.factory(self.context)
  File "/opt/python-3.6/lib/python3.6/site-packages/shapely/geometry/multipolygon.py", line 136, in geos_multipolygon_from_py
    assert L >= 1
AssertionError
>>> 
```

**Comment by sgillies** [comment]
Thanks for the script! This should be plenty to solve this.

**Comment by sgillies** [comment]
Resolved in 75598bd.
